tests/benchmark_api.py

```diff
@@ -0,0 +1,655 @@
+        Session.default_ontology = self.prev_default_ontology
+
+
+def benchmark_individual_classes(benchmark):
+    """Wrapper function for the IndividualClasses benchmark."""
+    return IndividualClasses.iterate_pytest_benchmark(
+        benchmark, size=DEFAULT_SIZE
+    )
+
+
+class IndividualConnect(Benchmark):
+    """Benchmark the `add` method, selecting the relationship."""
+
+    def _benchmark_set_up(self):
+        """Create a TBox and set it as the default ontology.
+
+        The new TBox contains SIMPHONY, OWL, RDFS and City.
+        """
+        self.ontology = Session(identifier="test-tbox", ontology=True)
+        self.ontology.load_parser(OntologyParser.get_parser("city"))
+        self.prev_default_ontology = Session.default_ontology
+        Session.default_ontology = self.ontology
+
+        from simphony_osp.namespaces import city
+
+        self.city_namespace = city
+
+        self.city = city.City(name="Freiburg", coordinates=[0, 0])
+        self.citizens = tuple(
+            city.Citizen(name=f"citizen {i}", age=25) for i in range(self.size)
+        )
+
+    def _benchmark_iterate(self, iteration=None):
+        self.city.connect(
+            self.citizens[iteration], rel=self.city_namespace.hasInhabitant
+        )
+
+    def _benchmark_tear_down(self):
+        """Restore the previous default TBox."""
+        Session.default_ontology = self.prev_default_ontology
+
+
+def benchmark_individual_connect(benchmark):
+    """Wrapper function for the IndividualAddRel benchmark."""
+    return IndividualConnect.iterate_pytest_benchmark(
+        benchmark, size=DEFAULT_SIZE
+    )
+
+
+class IndividualGetByIdentifier(Benchmark):
+    """Benchmark getting CUDS objects by uid (of type UUID)."""
+
+    def _benchmark_set_up(self):
+        """Create a TBox and set it as the default ontology.
+
+        The new TBox contains SIMPHONY, OWL, RDFS and City.
+        """
+        self.ontology = Session(identifier="test-tbox", ontology=True)
+        self.ontology.load_parser(OntologyParser.get_parser("city"))
+        self.prev_default_ontology = Session.default_ontology
+        Session.default_ontology = self.ontology
+
+        from simphony_osp.namespaces import city
+
+        self.city = city.City(name="Freiburg", coordinates=[0, 0])
+        self.citizens = tuple(
+            city.Citizen(name=f"citizen {i}", age=25) for i in range(self.size)
+        )
+        self.identifiers = tuple(
+            citizen.identifier for citizen in self.citizens
+        )
+        for citizen in self.citizens:
+            self.city.connect(citizen, rel=city.hasInhabitant)
+
+    def _benchmark_iterate(self, iteration: int = None):
+        self.city.get(self.identifiers[iteration])
+
+    def _benchmark_tear_down(self):
+        """Restore the previous default TBox."""
+        Session.default_ontology = self.prev_default_ontology
+
+
+def benchmark_individual_get_byidentifier(benchmark):
```

Comment:
```suggestion
def benchmark_individual_get_by_identifier(benchmark):
```

Reply:
Dankeschön

Reply:
Oops true, actually the initial intention was to write them like this I believe